### PR TITLE
Replace SequentialLR warmup with custom cosine wrapper

### PR DESCRIPTION
**Pull request by ShinDongWoon**
## Summary
- add a WarmupThenCosineScheduler wrapper that owns the warmup LinearLR and cosine decay schedulers on a shared optimizer
- switch the cosine warmup branch in the trainer to use the new wrapper and keep warmup-specific LR initialisation in sync
- ensure the wrapper serialises/deserialises both sub-schedulers alongside its own step counter

## Testing
- `python - <<'PY'
import sys
from pathlib import Path
sys.path.append(str(Path(__file__).resolve().parents[0] / 'src'))

import torch
import warnings
from torch.optim.lr_scheduler import LinearLR, CosineAnnealingLR
from timesnet_forecast.train import WarmupThenCosineScheduler

base_lr = 0.01
warmup_start_factor = 0.1
warmup_epochs = 3
T_max = 12
eta_min = 1e-4

param = torch.nn.Parameter(torch.tensor(1.0))
optim = torch.optim.SGD([param], lr=base_lr)
warmup = LinearLR(optim, start_factor=warmup_start_factor, end_factor=1.0, total_iters=warmup_epochs)
cosine = CosineAnnealingLR(optim, T_max=max(1, T_max - warmup_epochs), eta_min=eta_min)
scheduler = WarmupThenCosineScheduler(warmup, cosine, warmup_epochs)

initial_lr = base_lr * warmup_start_factor
for group in optim.param_groups:
    group['lr'] = initial_lr
scheduler._last_lr = [initial_lr for _ in scheduler._last_lr]
if hasattr(scheduler.warmup_scheduler, '_last_lr'):
    scheduler.warmup_scheduler._last_lr = [initial_lr for _ in scheduler.warmup_scheduler._last_lr]
if hasattr(scheduler.main_scheduler, '_last_lr'):
    scheduler.main_scheduler._last_lr = [initial_lr for _ in scheduler.main_scheduler._last_lr]

with warnings.catch_warnings(record=True) as caught:
    for _ in range(T_max):
        optim.step()
        scheduler.step()

assert not caught, f"Expected no warnings, got {[str(w.message) for w in caught]}"
print('final_lr', optim.param_groups[0]['lr'])
PY`

------
https://chatgpt.com/codex/tasks/task_e_68dc6ab268cc832883612f0a8a1e6ef4

### CODE REVIEW

**Comment by ShinDongWoon** [comment]
@codex review
